test/test_sparse_int_set.jl

```diff
@@ -0,0 +1,244 @@
+        @test isempty(intersect(SparseIntSet()))
+        @test isempty(intersect(SparseIntSet(1:10), SparseIntSet()))
+        @test isempty(intersect(SparseIntSet(), SparseIntSet(1:10)))
+        @test isempty(intersect(SparseIntSet(), complement(SparseIntSet())))
+        @test isempty(intersect(SparseIntSet(), complement(SparseIntSet(1:10))))
+        @test isempty(intersect(complement(SparseIntSet()), SparseIntSet()))
+        @test isempty(intersect(complement(SparseIntSet(1:10)), SparseIntSet()))
+
+        @test intersect(SparseIntSet([1,2,3])) == SparseIntSet([1,2,3])
+        @test intersect(complement!(SparseIntSet(1)), SparseIntSet(2)) ==
+              intersect(SparseIntSet(2), complement!(SparseIntSet(1))) == SparseIntSet(2)
+
+        @test intersect(SparseIntSet(1:7), SparseIntSet(3:10)) ==
+              intersect(SparseIntSet(3:10), SparseIntSet(1:7)) == SparseIntSet(3:7)
+        @test intersect(complement(SparseIntSet([1:2; 11:16])), SparseIntSet(1:7)) ==
+              intersect(SparseIntSet(1:7), complement(SparseIntSet([1:2; 11:16]))) == SparseIntSet(3:7)
+
+        @test intersect(complement(SparseIntSet(5:12)), complement(SparseIntSet(7:10))) ==
+              intersect(complement(SparseIntSet(7:10)), complement(SparseIntSet(5:12))) == complement(SparseIntSet(5:12))
+
+        @test intersect!(SparseIntSet(1:10), SparseIntSet(1:4), 1:5, [1,2,10]) == SparseIntSet(1:2)
+    end
+
+    @testset "Setdiff" begin
+        s1 = SparseIntSet(1:100)
+        setdiff!(s1, SparseIntSet(1:2:100))
+        s2 = setdiff(SparseIntSet(1:100), SparseIntSet(1:2:100))
+        @test s1 == s2 == SparseIntSet(2:2:100)
+
+        s1 = SparseIntSet(1:10)
+        s2 = complement(SparseIntSet(3:5))
+        @test setdiff(s1, s2) == setdiff(s1, [1:2; 6:100]) == SparseIntSet(3:5)
+        @test isempty(setdiff(complement(SparseIntSet()), complement(SparseIntSet())))
+        @test setdiff(complement(SparseIntSet(4)), complement(SparseIntSet(3:5))) == SparseIntSet((3,5))
+        @test setdiff(complement(SparseIntSet(1:5)), complement(SparseIntSet(3:10))) == SparseIntSet([6, 7, 8, 9, 10])
+        @test setdiff(complement(SparseIntSet(2:2:10)), SparseIntSet(1:5)) == complement(SparseIntSet([1:5; 6:2:10]))
+        @test setdiff!(complement(SparseIntSet(5)), complement(SparseIntSet(5))) == SparseIntSet()
+        @test setdiff!(complement(SparseIntSet(1:2:10)), complement(SparseIntSet(1:10))) == SparseIntSet(2:2:10)
+    end
+
+    @testset "Subsets, equality" begin
+        @test SparseIntSet(2:2:10) < SparseIntSet(1:10)
+        @test !(SparseIntSet(2:2:10) < SparseIntSet(2:2:10))
+        @test SparseIntSet(2:2:10) <= SparseIntSet(2:10)
+        @test SparseIntSet(2:2:10) <= SparseIntSet(2:2:10)
+        @test SparseIntSet(1) < complement!(SparseIntSet([5]))
+        @test SparseIntSet(1) <= complement!(SparseIntSet([20]))
+        @test !(SparseIntSet(1) < complement!(SparseIntSet(1)))
+    end
+
+    @testset "other 1" begin
```

Comment:
What is this?

Reply:
yea I literally straight copied it from the `test_int_set.jl` so not sure where that naming came from.